### DISCUSSION TITLE
Todos before v0.9

### DISCUSSION THREAD

**Discussion by lg-nineoneninethreenine** [sumeshir26/TimerX#10] [General]
So, we got a lot of tasks after the merge of #9, #8 had a big commit history, so no. 9 was created. It's simple, not a big commit history but big changelogs. In fact, you can view them [here]() and [here]().
Now, we need to do some tasks after the big merge(200 additions, 100 removals in #9).
They are as follows:

- [x] Merge #9.
- [x] Test on Windows 7 and 11 (@im-coder-lg and @sumeshir26, others are welcome to help on this)
- [x] Test on macOS High Sierra+ (@not-nef but can be done during the holidays as said in #6 or anyone else, just volunteer here!)
- [x] Make sure checks succeed without loose ends (@im-coder-lg #12)
- [x] Document changes (@im-coder-lg and @sumeshir26 for review)
- [x] Triage on installer bug([error code `2756`, comment linked](https://github.com/sumeshir26/TimerX/pull/8#issuecomment-989600371), upstream(cx_Freeze error) and will be ticked off after finding the fix.)
- [x] Get ready for v1.0.0-alpha!
I think I will document the changes on StackEdit and post it as a secret gist [here](https://gist.github.com/im-coder-lg/ba650b83d3e45c647050524e7fc3b90d).
These are the tasks to complete for v1.0.0-alpha's release. After that, we head towards beta, maybe RCs, and then hit the major v1.0.0 milestone.
Edit by @im-coder-lg: We could try using the Auto Release notes feature to make automatic notes. testing that now.
Edit by @im-coder-lg: We hit the jackpot. The Auto Release Notes feature is all we want! Read the entire section below:
<details>
<summary>Auto Release Notes</summary>
<section>

## What's Changed
* feat(including bugs): minor changes that increases functionality by @im-coder-lg in https://github.com/sumeshir26/TimerX/pull/1
* feat: Dark mode by @sumeshir26 in https://github.com/sumeshir26/TimerX/pull/2
* feat: Always on top by @sumeshir26 in https://github.com/sumeshir26/TimerX/pull/4
* fix: Actions by @sumeshir26 in https://github.com/sumeshir26/TimerX/pull/9
* Patch actions by @im-coder-lg in https://github.com/sumeshir26/TimerX/pull/12
* feat: Actions Improvements by @sumeshir26 in https://github.com/sumeshir26/TimerX/pull/13
* feat: Actions improvements(Part 2) by @sumeshir26 in https://github.com/sumeshir26/TimerX/pull/14
* feat: Unified action by @sumeshir26 in https://github.com/sumeshir26/TimerX/pull/19

## New Contributors
* @im-coder-lg made their first contribution in https://github.com/sumeshir26/TimerX/pull/1
* @sumeshir26 made their first contribution in https://github.com/sumeshir26/TimerX/pull/2
Adding one more thing:
* @not-nef made their first contribution in #18.

No changelogs in a demo.

</section>
</details>
Edit by @im-coder-lg: Can we create a separate category on Discussions just for release tasks? So that this is not a general talk, it's kinda important, right?

**Comment by sumeshir26** [comment]
@im-coder-lg, what features would you like to be added to TimerX? And do you actuallay use timerx?
What about the tar.gz?

**Reply by lg-nineoneninethreenine** [reply]
> what features would you like to be added to TimerX?

First, a settings tab.
Next, the ability to activate blur as an optional feature, kinda like an Easter egg, like Google Easter eggs, it will be hidden.
After that, a simple documentation site where we address common issues(I know how to manage this 😉)

> And do you actuallay use timerx?

I don't use it much since I don't keep a lot of timers, but I will try one and as the usual way, forget about it after 5 minutes. But it's be fun seeing my reaction after realising that TimerX was running, with a timer. Hopefully I'll get reminded to eat lunch after some time in my PC.

> What about the tar.gz?

That is a hard question. If we can package it, it'd be easy. But half of Google searches are dead ends. But I think I will find something for the tarball and work on it.

**Comment by sumeshir26** [comment]


**Reply by lg-nineoneninethreenine** [reply]
We need to document the (major) commits that happened since v0.1 so that's what I meant. I will document it(after #9 is merged), can you review it?

**Reply by sumeshir26** [reply]
@im-coder-lg Sure

**Comment by lg-nineoneninethreenine** [comment]
@sumeshir26 can you just trigger the build workflow? So that we can know if it succeeds here?

**Reply by sumeshir26** [reply]
Okay

**Reply by lg-nineoneninethreenine** [reply]
Now, the checks work. PS: The macOS environment uses is macOS Catalina(10.15), so we can confirm that this will work on Big Sur and Monterey. If @not-nef or anyone with a local Mac machine tests the installer and the app, we can tell that macOS works.

**Reply by sumeshir26** [reply]
@im-coder-lg Do the notifications work on Win7

**Reply by lg-nineoneninethreenine** [reply]
Yeah it shows up as a small bubble which is fine.

**Comment by sumeshir26** [comment]
@im-coder-lg 
![image](https://user-images.githubusercontent.com/68823982/145764434-4fe0420f-ac13-4a89-923b-c9009a6fe00f.png)
I tried to run `bdist_rpm` on replit, and it worked! It created a tar.gz at `dist/TimerX-[version].tar.gz!` So we _can_ use github actions for tar.gz!

**Comment by lg-nineoneninethreenine** [comment]
What's the command?
Also, in the cx-Freeze issue, I figured out something. @marcelotduarte told 
us to use Alien(sudo apt install alien) on Debian-based systems(that 
includes Ubuntu), so we could use that to package files on a Gitpod env.

**Reply by sumeshir26** [reply]
@im-coder-lg python setup.py bdist_rpm

**Reply by lg-nineoneninethreenine** [reply]
And then, it forms the tarball?

**Reply by lg-nineoneninethreenine** [reply]
I'm signing up on Replit and testing this, need to confirm myself. What is the OS Replit uses?

**Reply by lg-nineoneninethreenine** [reply]
Alright, Bionic Beaver! Ubuntu LTS, so we got the APT package manager. I'll manage this from here.

**Reply by lg-nineoneninethreenine** [reply]
Hmm... Replit doesn't seem right, it doesn't allow APT packages. Testing the same in Gitpod. Wonder if we could add Gitpod support here.

**Reply by lg-nineoneninethreenine** [reply]
Also, can you share your updated setup.py snippet? If the config exists, we could use RPM builds.

**Reply by sumeshir26** [reply]
@im-coder-lg You dont even need alien on replit, you just run the command

**Reply by lg-nineoneninethreenine** [reply]
Yeah, but to convert it into .DEB files, we can use `alien`. It literally listed this"
```console
$ alien --help
Usage: alien [options] file [...]
  file [...]                Package file or files to convert.
  -d, --to-deb              Generate a Debian deb package (default).
     Enables these options:
       --patch=<patch>      Specify patch file to use instead of automatically
                            looking for patch in /var/lib/alien.
       --nopatch            Do not use patches.
       --anypatch           Use even old version os patches.
       -s, --single         Like --generate, but do not create .orig
                            directory.
       --fixperms           Munge/fix permissions and owners.
       --test               Test generated packages with lintian.
  -r, --to-rpm              Generate a Red Hat rpm package.
      --to-slp              Generate a Stampede slp package.
  -l, --to-lsb              Generate a LSB package.
  -t, --to-tgz              Generate a Slackware tgz package.
     Enables these options:
       --description=<desc> Specify package description.
       --version=<version>  Specify package version.
  -p, --to-pkg              Generate a Solaris pkg package.
  -i, --install             Install generated package.
  -g, --generate            Generate build tree, but do not build package.
  -c, --scripts             Include scripts in package.
      --target=<arch>       Set architecture of the generated package.
  -v, --verbose             Display each command alien runs.
      --veryverbose         Be verbose, and also display output of run commands.
  -k, --keep-version        Do not change version of generated package.
      --bump=number         Increment package version by this number.
  -h, --help                Display this help message.
  -V, --version             Display alien's version number.
```
This is saying that we can create DEB, RPM and tarballs from that tool.

**Reply by lg-nineoneninethreenine** [reply]
Anyway, I'll test that.

**Reply by lg-nineoneninethreenine** [reply]
Well, test results time!
Firstly, Replit did the job in 1 millisecond! But, look at this:
![image](https://user-images.githubusercontent.com/76845820/145767633-543dd153-df54-4b41-8900-cff8b49ad667.png)
It's supposed to have some `/usr` directories, but it repackaged the `setup.py` file.
Logs:
```shell
~/TimerX$ ls dist/TimerX-1.0.0.0/* -la
-rw-r--r-- 1 runner runner 1067 Dec 13 06:49 dist/TimerX-1.0.0.0/LICENSE
-rw-r--r-- 1 runner runner  219 Dec 13 07:05 dist/TimerX-1.0.0.0/PKG-INFO
-rw-r--r-- 1 runner runner 1001 Dec 13 06:49 dist/TimerX-1.0.0.0/README.md
-rw-r--r-- 1 runner runner   38 Dec 13 07:05 dist/TimerX-1.0.0.0/setup.cfg
-rw-r--r-- 1 runner runner 1812 Dec 13 06:49 dist/TimerX-1.0.0.0/setup.py

dist/TimerX-1.0.0.0/TimerX.egg-info:
total 16
drwxr-xr-x 1 runner runner 104 Dec 13 07:05 .
drwxr-xr-x 1 runner runner 112 Dec 13 07:05 ..
-rw-r--r-- 1 runner runner   1 Dec 13 07:05 dependency_links.txt
-rw-r--r-- 1 runner runner 219 Dec 13 07:05 PKG-INFO
-rw-r--r-- 1 runner runner 146 Dec 13 07:05 SOURCES.txt
-rw-r--r-- 1 runner runner   1 Dec 13 07:05 top_level.txt
```
So, we'll test the `alien` tool.

**Reply by lg-nineoneninethreenine** [reply]
The main part is the command: `tar xzf TimerX-*.tar.gz`

**Reply by lg-nineoneninethreenine** [reply]
I'm trying this: https://www.devdungeon.com/content/debian-package-tutorial-dpkgdeb#toc-3

**Reply by sumeshir26** [reply]
Ok

**Reply by lg-nineoneninethreenine** [reply]
Oh I meant, "Let's try this", sorry for that, must've been in a hurry.
Also, I got an idea. Can you install Fedora WSL on your computer and build the ROM package on that? After that, can you make a zip file and send it here? I will test the `alien` tool for that.

**Reply by sumeshir26** [reply]
@im-coder-lg Fedora isnt on WSL☹️

**Reply by sumeshir26** [reply]
> build the ROM package on that? 

ROM or RPM?

**Reply by lg-nineoneninethreenine** [reply]
RPM, sorry about that. What about openSUSE?

**Reply by sumeshir26** [reply]
@im-coder-lg Doing that, can you look at the latest action run?

**Reply by lg-nineoneninethreenine** [reply]
Ubuntu fails from here: https://github.com/sumeshir26/TimerX/runs/4530058102?check_suite_focus=true#step:5:80
Can't we try building on Fedora/any RHEL-based systems like CentOS and then publish here?

**Reply by lg-nineoneninethreenine** [reply]
I wonder if we could install Fedora on Raspberry Pi...

**Comment by sumeshir26** [comment]
@im-coder-lg I did try getting OpenSUSE WSL but it gave some error when i ran pip install cx_freeze

**Reply by lg-nineoneninethreenine** [reply]
Did you try pip3 and sudo?

**Comment by sumeshir26** [comment]
I also posted a question on Stack Overflow, waifting on a response on that.
Also, can you paste the markdown for yojur md markdown design?

**Reply by lg-nineoneninethreenine** [reply]
Oops, didn't notice this... What markdown design? If you are asking for the banner one, it is this:
```md
| your-text
| ---
```
Preview:
| your-text
| ---

**Comment by sumeshir26** [comment]
Yes, those work, but cx-freseze fails to install

**Reply by lg-nineoneninethreenine** [reply]
I just got an idea, will inform later.

**Comment by sumeshir26** [comment]
@im-coder-lg Whats your idea

**Reply by lg-nineoneninethreenine** [reply]
I'mma try getting CentOS on my Pi. Get my peripherals, get the TV on, comment it, flash CentOS and then get a DE. After that, I'll test the RPM build and try compiling it to Debian soon.

**Reply by sumeshir26** [reply]
@im-coder-lg I fixed the cx_freeze issue. Now alien is failing because it requires elevation. Do you know how to fix this?

**Reply by lg-nineoneninethreenine** [reply]
`sudo` is best. Have you even tried `su`?

**Reply by sumeshir26** [reply]
@sumeshir26 Fixed using sudo already… Almost fixed… tomorrow it will be done…

**Reply by sumeshir26** [reply]
@im-coder-lg

**Reply by lg-nineoneninethreenine** [reply]
Sorry for that... Got a really bad headache so dozed off... I'll also test by installing CentOS, I just need some free time.

**Reply by sumeshir26** [reply]
@im-coder-lg Try this
[timerx_1.0-2_all.deb.zip](https://github.com/sumeshir26/TimerX/files/7738611/timerx_1.0-2_all.deb.zip)
:

**Reply by lg-nineoneninethreenine** [reply]
Testing that soon!

**Reply by lg-nineoneninethreenine** [reply]
Have you tried breaking the DEB down with PeaZip? I did and it seems to be confusing... Will post the screenshots soon here.

**Comment by sumeshir26** [comment]
@im-coder-lg I Fixed the MSI error, it should work now!

**Reply by lg-nineoneninethreenine** [reply]
Test complete, updating the todo list!

**Comment by lg-nineoneninethreenine** [comment]
Hey, just wanna tell you guys this(also @not-nef welcome to the Contributors community!) that the Ubuntu build in the final Release build kinda seems like it's malfunctioning. I have collapsed the logs below, so you can check that if you want.
<details>
<summary>Logs</summary>
<section>


```console
python setup.py bdist_rpm
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.8.12/x64
running bdist_rpm
creating build
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/rpm
creating build/bdist.linux-x86_64/rpm/SOURCES
creating build/bdist.linux-x86_64/rpm/SPECS
creating build/bdist.linux-x86_64/rpm/BUILD
creating build/bdist.linux-x86_64/rpm/RPMS
creating build/bdist.linux-x86_64/rpm/SRPMS
writing 'build/bdist.linux-x86_64/rpm/SPECS/TimerX.spec'
running sdist
running egg_info
creating TimerX.egg-info
writing TimerX.egg-info/PKG-INFO
writing dependency_links to TimerX.egg-info/dependency_links.txt
writing top-level names to TimerX.egg-info/top_level.txt
writing manifest file 'TimerX.egg-info/SOURCES.txt'
adding license file 'LICENSE' (matched pattern 'LICEN[CS]E*')
reading manifest file 'TimerX.egg-info/SOURCES.txt'
writing manifest file 'TimerX.egg-info/SOURCES.txt'
running check
warning: check: missing required meta-data: url
creating TimerX-1.0

warning: check: missing meta-data: either (author and author_email) or (maintainer and maintainer_email) must be supplied

creating TimerX-1.0/TimerX.egg-info
copying files to TimerX-1.0...
copying LICENSE -> TimerX-1.0
copying README.md -> TimerX-1.0
copying setup.py -> TimerX-1.0
copying TimerX.egg-info/PKG-INFO -> TimerX-1.0/TimerX.egg-info
copying TimerX.egg-info/SOURCES.txt -> TimerX-1.0/TimerX.egg-info
copying TimerX.egg-info/dependency_links.txt -> TimerX-1.0/TimerX.egg-info
copying TimerX.egg-info/top_level.txt -> TimerX-1.0/TimerX.egg-info
Writing TimerX-1.0/setup.cfg
creating dist
Creating tar archive
removing 'TimerX-1.0' (and everything under it)
copying dist/TimerX-1.0.tar.gz -> build/bdist.linux-x86_64/rpm/SOURCES
copying ./assets/logo.png -> build/bdist.linux-x86_64/rpm/SOURCES
building RPMs
error: Unable to open icon /home/runner/rpmbuild/SOURCES/logo.png: No such file or directory
error: query of specfile build/bdist.linux-x86_64/rpm/SPECS/TimerX.spec failed, can't parse
error: Failed to execute: "rpm -q --qf '%{name}-%{version}-%{release}.src.rpm %{arch}/%{name}-%{version}-%{release}.%{arch}.rpm\\n' --specfile 'build/bdist.linux-x86_64/rpm/SPECS/TimerX.spec'"
Error: Process completed with exit code 1.
```

</section>
</details>
Process failed, so maybe if GitHub has a Fedora Linux image or a CentOS image, we could try on that and then convert the entire RPM into DEB personally, right?

**Reply by lg-nineoneninethreenine** [reply]
Anyway, it's kinda solved. I think it's kind of pointless to have the DEB. Try getting PeaZip Portable, open the DEB on that and just check the so-called archive.

**Comment by lg-nineoneninethreenine** [comment]
People, I feel we are ready for the next release. @sumeshir26 can you please tell the release name 'cause you got my brain haywired in #21.
After that, let's release the next version on Christmas(your gift this year :wink:) or New Year and then maybe get the next version out by Febraury of 2022.

**Reply by sumeshir26** [reply]
The name is v1.0-alpha. Need to update setup.py soon

**Reply by lg-nineoneninethreenine** [reply]
I will add that to the PR of mine.

**Comment by lg-nineoneninethreenine** [comment]
Hey @sumeshir26 can you setup Vercel here? For the website ofc, but during the setup, change the root dir and change the framework to Docusaurus 2.

**Reply by sumeshir26** [reply]
Ok, wait till wednesday.

**Reply by lg-nineoneninethreenine** [reply]
> Ok, wait till wednesday.

Wait, what? PS: Setup takes only 5 minutes, you'll have to check if it's into production soon.
